dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/InternalLogger.kt

```diff
@@ -111,6 +111,7 @@ interface InternalLogger {
          * Logger for the cases when SDK instance is not yet available. Try to use the logger
          * provided by [FeatureSdkCore.internalLogger] instead if possible.
          */
+        @JvmField
         val UNBOUND: InternalLogger = SdkInternalLogger(null)
```

Comment:
I think you don't need to use `UNBOUND`. According to the code, you have access to the `sdkCore.internalLogger` in the `OtelTracerProvider.Builder.build()` which calls `CoreTracer.CoreTracerBuilder()` - you can just inject `sdkCore.internalLogger` there and pass it down to the `CoreTracer` instance then.

Is there a need in the `UNBOUND` then and thus this change?

Reply:
I'm not sure why it should be optional. If it is provided in the constructor of `CoreTracer.CoreTracerBuilder()` why it should be?

`InternalLogger.UNBOUND` is only for cases when there is no possibility to get logger associated with particular SDK core. This logger is not able to send telemetry and logs produced cannot print the SDK core instance used.

Reply:
yes...that's the thing, I did not want to provide it in the constructor, not sure why, can't remember, but think it about it now I don't see why I should not.

Reply:
actually it is because I do not want to use modify the static method: `CoreTracer:builder` to `CoreTracer:builder(logger)` but will provide a NoOpInternalLogger implementation instead.